app/src/main/java/io/appium/uiautomator2/server/AppiumServlet.java

```diff
@@ -161,6 +162,7 @@ private void registerPostHandler() {
 
     private void registerGetHandler() {
         register(getHandler, new Status("/status"));
+        register(getHandler,new Getpackages("/apps"));
```

Comment:
this is not a standard w3c endpoint, so we should not expose it like this. Consider adding this data into the existing /info endpoint instead

Reply:
For driver, we have to add in the driver Wrapper SDK right?

Reply:
I have changed the endpoint to /appium/device/apps

Reply:
does it make sense to have this endpoint without a session id?

Reply:
> For driver, we have to add in the driver Wrapper SDK right?

There should be a driver endpoint, which would expose this information. You could implement it in two ways, either using a `mobile: ...` extension or by adding a custom API endpoint and assume that clients would add a support for it.

Reply:
Example of a mobile extension: https://github.com/appium/appium-uiautomator2-driver/blob/7013ba401252a7de59c2fe15516440db4e606a4c/lib/commands/battery.js#L24

Example of a custom API endpoint: https://github.com/appium/appium-uiautomator2-driver/blob/7013ba401252a7de59c2fe15516440db4e606a4c/lib/method-map.js#L5

Reply:
Got it, i will do the necessary changes

Reply:
> does it make sense to have this endpoint without a session id?

Yes before starting a session, I should be able to get the application list for easy selection to start session on any of the listed application

Reply:
One cannot call any endpoint without having a session id. This is just how selenium protocol works. You need to call createSession first, get a unique id and then use it to call other APIs. No endpoint could be called prior to createSession.

So, how/where this implementation is supposed to be used by UIA2 driver?

Reply:
Correct, with driver we have to create session to utilize any api, from driver user has to create session to utilize the this API, i am utilizing the apis directly so i don't need to create session

Reply:
So if user is using diver than he can utilize after creating session

Reply:
I'm not quite convinced of adding an API, which is not supposed to be used by UIA2 driver then. You should fork the project if you use it in a custom way.

@KazuCocoa what do you think?

Reply:
I'm ok to add a new endpoint, if not a complicated feature to keep maintaining, but I'd request to add test code to check the function. Endpoints which are not in regular WebDriver protocol and not used by UIA2 driver regularly could break because we don't care so much for as our project. (Actually, fork is the most flexible way to maintain your own way as Mykola addressed.)

Reply:
Well i can maintain it, but i think it's a quite a useful feature for the End User to have and i think it can be easily maintained and should not affect the existing implementation

Reply:
I have already commented https://github.com/appium/appium-uiautomator2-driver/pull/585
It is just not going to work without a session id.

If you really need to have the endpoint in this rep I can propose to create two handlers for it - one with session id and one without (`session/:sessionId/appium/device/apps` and `/appium/device/apps` ). The latter could be then used in custom calls, but since the code would be shared, we can guarantee it works as expected.

Reply:
Okay makes sense, i will do that